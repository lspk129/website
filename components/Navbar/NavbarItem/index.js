import styled from 'styled-components';

const NavbarItem = styled.a`
  color: ${props => props.theme.text};
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 20px;
  cursor: pointer;
  display: inline-block;
  &:after {
    display: block;
    content: '';
    border-bottom: solid 3px ${props => props.theme.text};  
    transform: scaleX(0);  
    transition: transform 250ms ease-in-out;
  }
  &:hover:after {
    transform: scaleX(1);
  }

`;

export default NavbarItem;
