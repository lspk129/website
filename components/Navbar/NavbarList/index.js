import styled from 'styled-components';

const NavbarList = styled.div`
  text-align: center;
  position: fixed;
  grid-row: 2 / 3;
`;

export default NavbarList;
