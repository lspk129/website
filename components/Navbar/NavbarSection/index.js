import styled from 'styled-components';

const NavbarSection = styled.nav`
  width: 100%;
  background-color: ${props => props.theme.main};
  overflow: hidden;
  display: grid;
  grid-template-rows: 1fr;
  z-index: 10;
`;

export default NavbarSection;
