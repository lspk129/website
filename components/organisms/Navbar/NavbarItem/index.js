import styled from 'styled-components';

const NavbarItem = styled.div`
  color: ${props => props.theme.text};
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 16px;
  cursor: pointer;
  &:after {
    display: grid;
    content: '';
    border-bottom: solid 2px ${props => props.theme.text};  
    transform: scaleX(0);  
    transition: transform 250ms ease-in-out;
  }
  &:hover:after {
    transform: scaleX(1);
  }

`;

export default NavbarItem;
