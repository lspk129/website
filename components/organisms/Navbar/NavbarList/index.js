import styled from 'styled-components';

const NavbarList = styled.div`
  display: flex;
  align-items: center;
  text-align: center;
`;

export default NavbarList;
