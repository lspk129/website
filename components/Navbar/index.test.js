import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { StickyContainer, Sticky } from 'react-sticky';

import { Navbar, Logo } from './';
import NavbarSection from './NavbarSection';
import NavbarList from './NavbarList';
import NavbarItem from './NavbarItem';

describe('Navbar', () => {
  const shallowWrapper = mount(<StickyContainer><Navbar /></StickyContainer>);

  it('renders correctly', () => {
    const tree = toJson(shallowWrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders logo', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toBePresent();
  });

  it('renders NavbarSection', () => {
    expect(shallowWrapper).toBePresent();
    expect(shallowWrapper.find(NavbarSection)).toBePresent();
  });

  it('renders NavbarList', () => {
    expect(shallowWrapper.find(NavbarList)).toBePresent();
  });

  it('renders NavbarItem', () => {
    expect(shallowWrapper.find(NavbarItem)).toBePresent();
  });

  it('renders Sticky component', () => {
    expect(shallowWrapper.find(Sticky)).toBePresent();
  });
});
