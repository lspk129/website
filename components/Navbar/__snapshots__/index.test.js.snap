// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<Container>
  <div
    onScroll={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <Navbar>
      <Sticky
        bottomOffset={0}
        disableCompensation={false}
        disableHardwareAcceleration={false}
        relative={false}
        topOffset={0}
      >
        <div>
          <div />
          <styled.nav
            style={Object {}}
          >
            <nav
              className="sc-bdVaJa joTcTr"
              style={Object {}}
            >
              <styled.div
                className="NavbarList"
              >
                <div
                  className="NavbarList sc-htpNat fjwyka"
                >
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="sc-bxivhb etYArC"
                      onClick={[Function]}
                    >
                      Home
                    </div>
                  </styled.div>
                  <_
                    duration={500}
                    offset={0}
                    smooth={true}
                    spy={true}
                    to="Service"
                  >
                    <Link
                      onClick={[Function]}
                    >
                      <a
                        onClick={[Function]}
                      >
                        <styled.div>
                          <div
                            className="sc-bxivhb etYArC"
                          >
                            Services
                          </div>
                        </styled.div>
                      </a>
                    </Link>
                  </_>
                  <_
                    duration={500}
                    offset={0}
                    smooth={true}
                    spy={true}
                    to="Technology"
                  >
                    <Link
                      onClick={[Function]}
                    >
                      <a
                        onClick={[Function]}
                      >
                        <styled.div>
                          <div
                            className="sc-bxivhb etYArC"
                          >
                            Technologies
                          </div>
                        </styled.div>
                      </a>
                    </Link>
                  </_>
                  <styled.div
                    onClick={[Function]}
                  >
                    <div
                      className="sc-bxivhb etYArC"
                      onClick={[Function]}
                    >
                      Contact us
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </nav>
          </styled.nav>
        </div>
      </Sticky>
    </Navbar>
  </div>
</Container>
`;
